src/presentation/api/controller/virtualHost.go

```diff
@@ -0,0 +1,128 @@
+package apiController
+
+import (
+	"github.com/labstack/echo/v4"
+	internalDbInfra "github.com/speedianet/os/src/infra/internalDatabase"
+	apiHelper "github.com/speedianet/os/src/presentation/api/helper"
+	"github.com/speedianet/os/src/presentation/service"
+)
+
+type VirtualHostController struct {
+	virtualHostService *service.VirtualHostService
+}
+
+func NewVirtualHostController(
+	persistentDbSvc *internalDbInfra.PersistentDatabaseService,
+) *VirtualHostController {
+	return &VirtualHostController{
+		virtualHostService: service.NewVirtualHostService(persistentDbSvc),
+	}
+}
+
+// ReadVirtualHosts	 godoc
+// @Summary      ReadVirtualHosts
+// @Description  List virtual hosts.
+// @Tags         vhosts
+// @Security     Bearer
+// @Accept       json
+// @Produce      json
+// @Success      200 {array} entity.VirtualHost
+// @Router       /v1/vhosts/ [get]
+func (controller *VirtualHostController) Read(c echo.Context) error {
+	return apiHelper.ServiceResponseWrapper(c, controller.virtualHostService.Read())
+}
+
+// CreateVirtualHost    godoc
+// @Summary      CreateVirtualHost
+// @Description  Create a new vhost.
+// @Tags         vhosts
+// @Accept       json
+// @Produce      json
+// @Security     Bearer
+// @Param        createVirtualHostDto 	  body    dto.CreateVirtualHost  true  "Only hostname is required.<br />type may be 'top-level', 'subdomain', 'wildcard' or 'alias'. If is not provided, it will be 'top-level'. If type is 'alias', parentHostname it will be required."
```

Comment:
'parentHostname' will be required. No "it" in there and you forgot the single quotes.

Reply:
Done!